components/ILIAS/Mail/src/Cron/ScheduledMailsCron.php

```diff
@@ -0,0 +1,203 @@
+                $mailer = $this->umail
+                    ->withContextId(ilContext::CONTEXT_CRON);
+
+                $mailer->setSaveInSentbox(true);
+
+                $mailer->autoresponder()->enableAutoresponder();
+                $errors = $mailer->enqueue(
+                    $mail['rcp_to'],
+                    $mail['rcp_cc'],
+                    $mail['rcp_bcc'],
+                    $mail['m_subject'],
+                    $mail['m_message'],
+                    unserialize($mail['attachments'], ['allowed_classes' => false]),
+                    (bool) ($mail['use_placeholders'] ?? false)
+                );
+
+                if (empty($errors)) {
+                    $sent_mail_ids[] = (int) ($mail['mail_id'] ?? 0);
+                }
+            } catch (Exception $e) {
+                $job_result->setStatus(JobResult::STATUS_FAIL);
+                ilLoggerFactory::getLogger('mail')->error(
+                    'Error sending scheduled mail with id ' . ($mail['mail_id'] ?? 'unknown') . ': ' . $e->getMessage()
+                );
+                $job_result->setMessage($e->getMessage());
+
+                return $job_result;
+            }
+        }
+        $this->mail->deleteMails($sent_mail_ids);
+        ilLoggerFactory::getLogger('mail')->info(
+            'Sent ' . count($sent_mail_ids) . ' scheduled mails and removed them from outbox.'
+        );
+        $job_result->setMessage('Processed ' . count($sent_mail_ids) . ' mails.');
+
+        return $job_result;
+    }
+
+    /**
+     * @return Generator<list{
+     *     mail_id: int,
+     *     rcp_to: string,
+     *     rcp_cc: string,
+     *     rcp_bcc: string,
+     *     m_subject: string,
+     *     m_message: string,
+     *     attachments: string,
+     *     use_placeholders: bool,
+     *     schedule_datetime: string,
+     *     schedule_timezone: string
+     * }>
+     */
+    public function getOutboxMails(): Generator
+    {
+        $res = $this->db->queryF(
+            <<<'SQL'
+            SELECT 
+                mail_id, 
+                rcp_to, 
+                rcp_cc, 
+                rcp_bcc, 
+                m_subject,
+                m_message, 
+                attachments, 
+                use_placeholders, 
+                schedule_datetime, 
+                schedule_timezone
+            FROM mail 
+            JOIN mail_obj_data ON mail.folder_id = mail_obj_data.obj_id 
+                 WHERE mail_obj_data.m_type = %s 
+                   AND schedule_datetime IS NOT NULL
+            SQL,
+            [ilDBConstants::T_TEXT],
+            ['outbox']
+        );
+        $current_timestamp = time();
+
+        while ($row = $this->db->fetchAssoc($res)) {
+            $schedule_datetime = new DateTimeImmutable(
+                $row['schedule_datetime'],
+                new DateTimeZone($row['schedule_timezone'])
+            );
+            if ($schedule_datetime->getTimestamp() <= $current_timestamp) {
```

Comment:
IMO it should be enough to compare the two variables, we don't need to call `getTimestamp`.